### DISCUSSION TITLE
Storage object filename with multiple dot

### DISCUSSION THREAD

**Discussion by 0x67** [supabase/supabase#7464] [Questions] [answered]
My storage policy 
```
((bucket_id = 'resources'::text) AND (storage.extension(name) = 'pdf'::text) AND (lower((storage.foldername(name))[1]) = 'public'::text) AND (role() = 'anon'::text))
```

My upload code - `Working`
```
await this.supabaseClient.storage
	.from(bucket)
    .upload('public/userId/randomId/fileName.pdf', file.buffer, {
	    contentType: file.mimetype,
    });
 ```

So my project require me to allow user to modify filename, I achieve it like this - `Working`
```
await  this.bucketService.storage
	.from(bucket)
	.move(`public/userId/randomId/fileName.pdf`,`public/userId/randomId/newFileName.pdf');
```
But if I change the `newFileName.pdf` to `new.file.Name.pdf` it returns error
`new row violates row-level security policy for table \"objects\"`

However, if uploading via UI with the same file (`new.file.Name.pdf`) it's working without error. 

Can anyone pointing out what I'm doing wrong here?

Edit: the client is instantiated using anon key

**Comment by GaryAustin1** [comment, marked as answer]
At one time many many months ago the extension function was fixed to deal with multiple periods. It looks correct in the current version.  If that did not work you would get your permission error, but it would work in the UI.

**Reply by 0x67** [reply]
I'm a bit confused about this part. 

> It looks correct in the current version. If that did not work you would get your permission error, but it would work in the UI.

Does that mean my storage policy "looks" correct correct for the current version, but since doesn't work via client then did I misconfigure the policy or the UI use some additional configuration?

**Reply by GaryAustin1** [reply]
Sorry.  What you show looks correct to me :
(storage.extension(name) = 'pdf'::text

Should now work with SB storage code.  It would not have worked 6 months or more ago (I don't remember the exact time of the change).  They fixed it to deal with multiple periods...
https://github.com/supabase/storage-api/blob/c8b1053dacec0989c8e8189b6145eb1780e217a4/src/test/db/02-storage-schema.sql

So I don't know why/how you could have an old storage code base on your instance, if that is truly the problem.

You could take that line out of your policy and see if it works to verify that is the issue.  The fact you claim you can do the operation from your code with a different name says you have both select and update policy, which is required for move.

**Reply by 0x67** [reply]
Hmm my storage policy is generated today from the UI template actually
![image](https://user-images.githubusercontent.com/69191110/176203452-eb997f9c-0a62-4f4b-b47c-a340bda2af4a.png)

But thank you it solve my problem! I do have middleware to check for file extension, so I think it's okay not to check at the database level
